pkg/kubelet/kubelet_node_status.go

```diff
@@ -394,6 +398,29 @@ func (kl *Kubelet) updateNodeStatus() error {
 // tryUpdateNodeStatus tries to update node status to master. If ReconcileCBR0
 // is set, this function will also confirm that cbr0 is configured correctly.
 func (kl *Kubelet) tryUpdateNodeStatus(tryNumber int) error {
+	cancelFuncs := []context.CancelFunc{}
+	defer func() {
+		for _, cancelFunc := range cancelFuncs {
+			cancelFunc()
+		}
+	}()
+
+	// default to the direct v1 core client
+	heartbeatClient := kl.kubeClient.CoreV1()
+	// if possible, decorate requests to disable rate limiting and set a timeout
+	if restclient := heartbeatClient.RESTClient(); restclient != nil && !reflect.ValueOf(restclient).IsNil() {
+		heartbeatClient = v1core.New(&requestDecoratingRESTClient{
+			client: restclient,
+			decorator: func(req *rest.Request) *rest.Request {
+				ctx, cancelFunc := context.WithTimeout(context.Background(), kl.nodeStatusUpdateFrequency)
+				cancelFuncs = append(cancelFuncs, cancelFunc)
```

Comment:
Not sure it matters in the context of the current usage, but the decorated client here isn't thread-safe. Maybe worth a comment, at least?